### PR TITLE
Drop the `charge_taxes_on_shipping` flag from site settings.

### PR DESCRIPTION
**Pull request by maarcingebala**
In the new approach we don't use the global `charge_taxes_on_shipping` flag, instead, we check if there is a tax class assigned to the shipping method and use this tax class's rate.
 - For Avalara, we keep getting the shipping code from the plugin config, but we change the way to determine whether the tax should be added based on the above assumption.
 - Migration that creates tax classes and tax configuration for channels was split in two separate ones.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6794 Add logic to use flat tax rates in Saleor calculations](https://app.clickup.com/t/2549495/SALEOR-6794)

**Comment by patrys** [comment]
Task linked: [SALEOR-6391 Simple taxes](https://app.clickup.com/t/2549495/SALEOR-6391)

**Review by IKarbowiak — APPROVED** [review]

File: saleor/tax/migrations/0004_migrate_tax_classes.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            name = avatax_description or avatax_code
+            metadata = {
+                AVATAX_CODE_META_KEY: avatax_code,
+                AVATAX_DESCRIPTION_META_KEY: avatax_description,
+            }
+        if vatlayer_code:
+            name = vatlayer_code
+            metadata = {VATLAYER_CODE_META_KEY: vatlayer_code}
+
+    return name, metadata
```
Comment (korycins) [inline saleor/tax/migrations/0004_migrate_tax_classes.py:59]: If both codes are not None, you're using avatax code. In that case you don't need to have `if else {if & if}` 
```suggestion
    if avatax_code:
        name = avatax_description or avatax_code
        metadata = {
            AVATAX_CODE_META_KEY: avatax_code,
            AVATAX_DESCRIPTION_META_KEY: avatax_description,
        }
    elif vatlayer_code:
        name = vatlayer_code
        metadata = {VATLAYER_CODE_META_KEY: vatlayer_code}
```

Comment (maarcingebala) [reply saleor/tax/migrations/0004_migrate_tax_classes.py:59]: Yep, thanks Maciej ;)

**Review by korycins — APPROVED** [review]

**Pull request merged**